x-pack/plugins/reporting/server/lib/tasks/monitor_reports.ts

```diff
@@ -70,43 +64,58 @@ export class MonitorReportsTask implements ReportingTask {
     });
   }
 
+  /*
+   * This task finds report jobs that can get lost from the queue and must be recovered.
+   *
+   * The runner function reschedules a single report job per task run, to avoid
+   * flooding Task Manager in case many report jobs need to be recovered.
+   *
+   * Elasticsearch can send with a version conflict response when resetting the
+   * job status, in the event another instance of Kibana attempted to recover
+   * the same report job. That is normal.
+   */
   private getTaskRunner(): TaskRunCreatorFunction {
     return () => {
       return {
         run: async () => {
           const reportingStore = await this.getStore();
 
           try {
-            const results = await reportingStore.findZombieReportDocuments();
-            if (results && results.length) {
-              this.logger.info(
-                `Found ${results.length} reports to reschedule: ${results
-                  .map((pending) => pending._id)
-                  .join(',')}`
-              );
-            } else {
-              this.logger.debug(`Found 0 pending reports.`);
+            const recoveredJob = await reportingStore.findStaleReportJob();
+            if (!recoveredJob) {
+              // no reports need to be rescheduled
               return;
             }
 
-            for (const pending of results) {
```

Comment:
I didn't much like having a loop that handles multiple pending/expired reports. It easily caused things to go haywire.